### ISSUE TITLE
Speed up `premium_tax_credit` computation

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/openfisca-us#841]
@martinholmer has found that [`premium_tax_credit`](https://github.com/PolicyEngine/openfisca-us/blob/master/openfisca_us/variables/gov/irs/credits/premium_tax_credit/premium_tax_credit.py) currently takes a long time to run, because it's looking up everyone's second-lowest silver plan cost by state/area/age. Are there any opportunities to speed this up? Here are some ideas, largely trading off the ability to manipulate SLSPC as a parameter:
1. Assign SLSPC in the microdata generation (only helps for microsim, not individual)
2. Move the SLSPC from a parameter to a file like HDF5 or parquet, and join instead of looking up (no idea if this would help)
3. Apply numpy magic to bypass the SLSPC computation for ineligible tax units, e.g. something like [`np.sqrt(..., where=)`](https://stackoverflow.com/a/52631549/1840471) (no idea if this would work)

In the meantime, some computations can be sped up by setting `premium_tax_credit` to some value (e.g. zero) in tests or `IndividualSim` runs.

**Comment by martinholmer** [comment]
In my testing work, I typically execute YAML test files containing 1,000 to 10,000 households.  When the PTC logic was added recently, test runs time increased by a large amount.  When I used the `--performance-graph` test option, I found that the `slspc` calculation was taking almost 75% of the test execution time.  

I have been able to avoid these long run times by doing two things:
(a) adding a `has_marketplace_health_coverage: false` statement to each person
(b) making the following local change to the `premium_tax_credit` formula

```
from openfisca_us.model_api import *


class premium_tax_credit(Variable):
    value_type = float
    entity = TaxUnit
    label = "Premium Tax Credit"
    unit = USD
    definition_period = YEAR
    reference = "https://www.law.cornell.edu/uscode/text/26/36B"

    def formula(tax_unit, period, parameters):
        eligible = tax_unit("is_ptc_eligible", period)
        if all(~eligible):
            return eligible.astype(np.float32)
        plan_cost = tax_unit("second_lowest_silver_plan_cost", period)
        income = tax_unit("medicaid_income", period)
        applicable_percentage = tax_unit("ptc_phase_out_rate", period)
        return eligible * max_(0, plan_cost - income * applicable_percentage)
```

@MaxGhenis @nikhilwoodruff

**Comment by nikhilwoodruff** [comment]
Thanks @martinholmer for this - your local edit seems like a good idea to add in regardless, so I've filed #843. The other ideas also outlined by @MaxGhenis:
1. Yes, I think we should definitely do this.
2. Interesting, probably worth a try but not sure how much they'd increase speed.
3. Also worth a try, probably more likely to work than 2.

**Comment by nikhilwoodruff** [comment]
This should now be fixed by #858
